### PR TITLE
Support markdown for header attr in dropdown

### PR DESCRIPTION
**Pull request by le0tan**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] New feature

**What is the rationale for this request?**

In response to https://github.com/MarkBind/markbind/pull/1013#pullrequestreview-354233446

**What changes did you make? (Give an overview)**

Support markdown syntax for `header` attribute of a dropdown component. Let `text` attribute take priority over `header` (i.e. if `text` attribute is present, its value would be rendered as raw HTML in dropdown while `header` attribute, if present, would be ignored completely.)

**Testing instructions:**

Replace `vue-strap` from [this PR](https://github.com/MarkBind/vue-strap/pull/128)
Here are some test examples:
`<dropdown header="**strong**"></dropdown>`
`<dropdown header="alt text"></dropdown>`
`<dropdown text="<i>italic</i>" header="**s**"></dropdown>` - to test that `text` takes priority over `header`

**Proposed commit message: (wrap lines at 72 characters)**

Support markdown for header attribute in dropdown.
* header attr now supports inline markdown syntax
* text attr is reserved, and still supports raw HTML
* text attr now takes priority over header

### CODE REVIEW

**Comment by yamgent** [comment]
I think you forgot to update the test files.

File: src/lib/markbind/src/parsers/componentParser.js [hunk 1]
```diff
@@ -181,6 +181,29 @@ function _parseBoxAttributes(element) {
...
+ * Dropdowns
+ */
+
+function _parseDropdownAttributes(element) {
+  const el = element;
+  const slotChildren = el.children && el.children.filter(child => _.has(child.attribs, 'slot'));
+  const hasHeaderSlot = slotChildren && slotChildren.some(child => child.attribs.slot === 'header');
+
+  // If header slot is present, the header attribute has no effect, and we can simply remove it.
+  if (hasHeaderSlot) {
```
Comment (ang-zeyu) [inline src/lib/markbind/src/parsers/componentParser.js]: Perhaps we could delete `el.attribs.text` here, try to add deprecation notes for `text` where appropriate as well

File: src/lib/markbind/src/parsers/componentParser.js [hunk 2]
```diff
@@ -181,6 +181,29 @@ function _parseBoxAttributes(element) {
...
+    delete el.attribs.header;
+    return;
+  }
+
+  // header attribute takes priority over text attribute
+  if (_.has(element.attribs, 'header')) {
+    _parseAttributeWithoutOverride(element, 'header', true, '_header');
```
Comment (ang-zeyu) [inline src/lib/markbind/src/parsers/componentParser.js]: `_parseAttributeWithoutOverride` will not override the previous existing `_header` slot, and will delete the attribute.

Remember to delete `el.attribs.text` here as well! Or you could just do away with the `if .. else`

File: test/unit/utils/componentParserData.js [hunk 3]
```diff
@@ -251,4 +251,58 @@ module.exports.PARSE_BOX_ICON_EXPECTED = `
 </box>
 `;
 
+/**
+ * Dropdowns
+ */
+
+module.exports.PARSE_DROPDOWN_HEADER_EXPECTED = `
```
Comment (ang-zeyu) [inline test/unit/utils/componentParserData.js]: Small nit: perhaps move the `EXPECTED` constants below the constants they test to conform with the rest of the file

Could add deprecation notes here as well

File: docs/userGuide/components/advanced.md [hunk 4]
```diff
@@ -90,6 +90,10 @@ Slot name | Default class
 header <hr style="margin-top:0.2rem; margin-bottom:0" /> <small>`title` <br> (deprecated)</small>  | `popover-header`
 content | `popover-body` 
 
+**Dropdown Slot Options:**
+Slot name |
+--- | 
+header |
```
Comment (ang-zeyu) [inline docs/userGuide/components/advanced.md]: LGTM 👍 , just one final nit here, do include the default class applied on the direct parent ( I believe its `dropdown-toggle` )

**Comment by yamgent** [comment]
@le0tan Resolve the merge conflict

**Comment by le0tan** [comment]
@yamgent I resolved the conflicts.

**Comment by damithc** [comment]
Works in production. Good work @le0tan and reviewers.

**Pull request merged**